numba/np/random/old_distributions.py

```diff
@@ -390,20 +391,37 @@ def random_geometric(bitgen, p):
         return random_geometric_inversion(bitgen, p)
 
 
-@register_jitable
-def random_zipf(bitgen, a):
-    am1 = a - 1.0
-    b = pow(2.0, am1)
-    while 1:
-        U = 1.0 - next_double(bitgen)
-        V = next_double(bitgen)
-        X = np.floor(pow(U, -1.0 / am1))
-        if (X > INT64_MAX or X < 1.0):
-            continue
+if numpy_version < (2, 1):
+    @register_jitable
+    def random_zipf(bitgen, a):
+        am1 = a - 1.0
+        b = pow(2.0, am1)
+        while 1:
+            U = 1.0 - next_double(bitgen)
+            V = next_double(bitgen)
+            X = np.floor(pow(U, -1.0 / am1))
+            if (X > INT64_MAX or X < 1.0):
```

Comment:
Note: NumPy defines `RAND_INT_MAX` as `INT64_MAX` in the case of the non-legacy algorithm.